src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -451,3 +468,41 @@ BuildDistinctPlan(Query *masterQuery, Plan *subPlan)
 
 	return distinctPlan;
 }
+
+
+/*
+ * IsGroupBySubsetOfDistinct checks whether each clause in group clauses also
+ * exists in the distinct clauses.
+ */
+static bool
+IsGroupBySubsetOfDistinct(Query *masterQuery)
+{
+	List *distinctClauses = masterQuery->distinctClause;
+	List *groupClauses = masterQuery->groupClause;
+	ListCell *distinctCell = NULL;
+	ListCell *groupCell = NULL;
+
+	foreach(groupCell, groupClauses)
+	{
+		SortGroupClause *groupClause = (SortGroupClause *) lfirst(groupCell);
+		bool isFound = false;
+
+		foreach(distinctCell, distinctClauses)
+		{
+			SortGroupClause *distinctClause = (SortGroupClause *) lfirst(distinctCell);
+
+			if (groupClause->tleSortGroupRef == distinctClause->tleSortGroupRef)
+			{
+				isFound = true;
```

Comment:
Can't we remove `isFound` variable at all, and return directly? It makes slightly harder to follow.